collision_detection_fcl/src/collision_robot_fcl.cpp

```diff
@@ -40,14 +40,25 @@ collision_detection::CollisionRobotFCL::CollisionRobotFCL(const robot_model::Rob
   : CollisionRobot(model, padding, scale)
 {
   const std::vector<const robot_model::LinkModel*>& links = robot_model_->getLinkModelsWithCollisionGeometry();
+  std::size_t index;
   geoms_.resize(robot_model_->getLinkGeometryCount());
+  fcl_objs_.resize(robot_model_->getLinkGeometryCount());
   // we keep the same order of objects as what RobotState *::getLinkState() returns
   for (std::size_t i = 0 ; i < links.size() ; ++i)
     for (std::size_t j = 0 ; j < links[i]->getShapes().size() ; ++j)
     {
       FCLGeometryConstPtr g = createCollisionGeometry(links[i]->getShapes()[j], getLinkScale(links[i]->getName()), getLinkPadding(links[i]->getName()), links[i], j);
       if (g)
-        geoms_[links[i]->getFirstCollisionBodyTransformIndex() + j] = g;
+      {
+        index = links[i]->getFirstCollisionBodyTransformIndex() + j;
+        geoms_[index] = g;
+
+        // Need to store the FCL object so the AABB does not get recreated every time.
+        // Every time this object is created, g->computeLocalAABB() is called  which is
+        // very expensive and should only be calculated once. To update the AABB, use the
+        // collObj->setTsetTransform and then call collObj->computeAABB() to transform the AABB.
```

Comment:
Sorry, I have fixed the misspelling, but I have not pushed the changes yet. I was waiting to see if there were any addition modification required.